creator/data_templates/models.py

```diff
@@ -66,14 +72,111 @@ class Meta:
     )
     description = models.TextField(
         max_length=10000,
-        null=True,
-        blank=True,
         help_text="Description of data template",
     )
     icon = models.CharField(
         max_length=256,
         null=True,
         blank=True,
         help_text="Name of the Font Awesome icon to use when displaying the "
-        "template"
+        "template in the frontend web application"
     )
+
+    @property
+    def released(self):
+        """
+        Determines whether the data template is being used in any studies
+
+        A data template is being used in a study if one or more of its
+        versions has been assigned to a study
+        """
+        assigned_to_studies = (
+            TemplateVersion.objects
+            .filter(data_template__pk=self.pk)
+            .exclude(studies=None).count()
+        )
+        return assigned_to_studies > 0
+
+
+class TemplateVersion(models.Model):
+    """
+    A version of a specific data submission template
+
+    Encapsulates the data template's field definitions (e.g. columns,
+    accepted values, instructions for populating, etc.)
+    """
+
+    class Meta:
+        permissions = [
+            ("list_all_templateversion", "Show all template_version"),
```

Comment:
should `"Show all template_version"` be `show_all_template_version`?

Reply:
No bc its the the description of the permission so I think its ok to leave it in a human friendly form. Plus it follows the pattern that all of our permissions use